IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m

```diff
@@ -98,13 +98,44 @@ - (BOOL)updateAuthorityURL:(NSURL *)cacheAuthorityURL
                                        context:context error:error];
 }
 
-- (BOOL)clearForHomeAccountId:(NSString *)homeAccountId
-                     clientId:(NSString *)clientId
-                      context:(id<MSIDRequestContext>)context
-                        error:(NSError **)error
+- (BOOL)signedOutStateForHomeAccountId:(NSString *)homeAccountId
+                              clientId:(NSString *)clientId
+                               context:(id<MSIDRequestContext>)context
+                                 error:(NSError **)error
 {
+    if ([NSString msidIsStringNilOrBlank:homeAccountId]
+        || [NSString msidIsStringNilOrBlank:clientId])
+    {
+        if (error) *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidInternalParameter, @"Both homeAccountId and clientId are needed to query signed out state!", nil, nil, nil, nil, nil);
+        return NO;
+    }
+    
     MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWitHomeAccountId:homeAccountId clientId:clientId];
-    return [_metadataCache removeAccountMetadataForKey:key context:context error:error];
+    MSIDAccountMetadataCacheItem *acountMetadata = [_metadataCache accountMetadataWithKey:key context:context error:error];
+    if (!acountMetadata) { return NO; }
+    
+    return acountMetadata.isSignedOut;
+}
+
+- (BOOL)markSignedOutStateForHomeAccountId:(NSString *)homeAccountId
```

Comment:
does it mean you'd need to query account metadata for each account when reading all accounts?

Reply:
Good question!
To address such an issue, I added `[MSIDAccountMetadataCacheAccessor loadAccountMetadataForClientId: context: error:]` which will load all the relevant account metadata into memory in one keychain query.